File: dart-pass/q1-solution.dart
List listA = [
  "Muklah",
  "Diyar",
  "Noorhan",
  "Ali",
  "Abdulbasit",
  "Dena",
  "Saba",
  "Fadi",
  "Karrar",
  "Ahmed"
];
List listB = [
  "Diyar",
  "Noorhan",
  "Muklah",
  "Saba",
  "Mustafa",
  "Ahmed",
  "Fadi",
  "Dena",
  "Hassan",
  "Ali",
];

List getCommonNames(List listA, List listB){
  Set s1 = Set.from(listA);
  Set s2 = Set.from(listB);
  List result = s1.intersection(s2).toList();
  return result;
}

List getNamesContainM(List names){
  List result = [];
  final regex = RegExp(r"[mM]");
  
  for (String name in names){
    if (regex.hasMatch(name)){
      result.add(name);
    }
  }
  
  return result;
}

void main() {
  List commonNames = getCommonNames(listA, listB);
  List namesContainM = getNamesContainM(commonNames);

  print("The common names between listA and listB.");
  print(commonNames);
  print("----------");
  print("The names that contain the letter M in the common names.");
  print(namesContainM);
}

File: python-pass/q1-solution.py
def isolate(students):
    result = {}
    
    for name, grade in students.items():
        if (grade) not in result:
            result[grade] = [name]
        else:
            result[grade].append(name)

    return result


students = {
    'Ahmed': 'Class 1',
    'Mustafa': 'Class 3',
    'Ali': 'Class 2',
    'Sara': 'Class 1',
    'Zainab': 'Class 1',
    'Zain': 'Class 2',
    
}   
print(isolate(students))